src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,201 @@
+
+        if (fields.length != 2) {
+            throw new DukeException("Wrong format entered! Please enter <Event Name> /at <Event Date>");
+        }
+
+        if (fields[0].strip().equals("")) {
+            throw new DukeException("Event Name is empty!");
+        }
+
+        if (fields[1].strip().equals("")) {
+            throw new DukeException("No Date Specified!");
+        }
+
+        String taskName = fields[0].strip();
+        String dateString = fields[1].strip();
+        LocalDateTime date = Parser.parseDateTime(dateString);
+
+        if (date == null) {
+            return new Event(taskName, dateString);
+        }
+
+        return new Event(taskName, date);
+    }
+    private static Deadline newDeadline(String s) throws DukeException {
+        String[] fields = s.replaceFirst("deadline", "").split("/by");
+
+        if (fields.length != 2) {
+            throw new DukeException("Wrong format entered! Please enter <Deadline Name> /by <Deadline>");
+        }
+
+        if (fields[0].strip().equals("")) {
+            throw new DukeException("Deadline Name is empty!");
+        }
+
+        if (fields[1].strip().equals("")) {
+            throw new DukeException("No Date Specified!");
+        }
+
+        String taskName = fields[0].strip();
+        String dateString = fields[1].strip();
+        LocalDateTime date = Parser.parseDateTime(dateString);
+
+        if (date == null) {
+            return new Deadline(taskName, dateString);
+        }
+
+        return new Deadline(taskName, date);
+    }
+
+    public static String parseToFileFromTask(Task t) {
+        String dateStr = "";
+        if (t.getDateObj() == null) {
+            dateStr = t.getDate();
+            if (dateStr.equals("")) {
+                dateStr = "None";
+            }
+        } else {
+            DateTimeFormatter format = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+            dateStr = t.getDateObj().format(format);
+        }
+        return String.format("%c\t%c\t%s\t%s", t.getType(), t.getDone(), t.getTaskName(), dateStr);
+    }
+
+    public static Task parseToTaskFromFile(String fileInput) throws DukeException {
+        // <type>\t<done>\t<name>\t<date>
+        char type;
+        char done;
+        String name;
+        String dateStr;
+        LocalDateTime date;
+        Task t;
+
+        try {
+            String[] split = fileInput.split("\t");
+            type = split[0].toCharArray()[0];
+            done = split[1].toCharArray()[0];
+            name = split[2].strip();
+            dateStr = split[3];
+            date = Parser.parseDateTime(split[3]);
+        } catch (IndexOutOfBoundsException exception) {
+            throw new DukeException("Unable to load task from file!");
+        }
+
+        if (name.equals("")) {
+            throw new DukeException("Unable to load task from file!");
+        }
+
+
+        switch(type) {
+        case 'T':
```

Comment:
Good job on following the coding standard with regards to indentation for switch and case!